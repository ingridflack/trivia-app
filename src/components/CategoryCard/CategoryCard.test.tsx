import { MemoryRouter } from "react-router-dom";
import { CategoryCard } from ".";
import { describe, expect, it } from "vitest";
import { render } from "@testing-library/react";

describe("CategoryCard", () => {
  it("should match the snapshot", () => {
    const title = "Test Category";
    const icon = "test-icon.png";
    const color = "blue";

    const { container } = render(
      <MemoryRouter>
        <CategoryCard title={title} icon={icon} color={color} />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });
});
