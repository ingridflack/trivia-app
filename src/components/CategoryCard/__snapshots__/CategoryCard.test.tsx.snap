// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CategoryCard > should match the snapshot 1`] = `
<div>
  <div
    class="chakra-card css-1ws3iy8"
  >
    <div
      class="css-mssj5r"
    >
      <img
        alt="Test Category"
        class="chakra-image css-0"
        src="test-icon.png"
      />
    </div>
    <p
      class="chakra-text css-1ey7oos"
    >
      Test Category
    </p>
    <button
      class="chakra-button css-1k532n6"
      type="button"
    >
      Play
    </button>
  </div>
</div>
`;
